lib/arrays.md

```diff
@@ -5,44 +5,86 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is an Array?
 
-Replace me with your response...
+An array is a data object that stores an ordered list of values. 
 
 # What are some examples of information that would be Arrays as opposed to some other data type?
 
-Replace me with your response...
+Arrays can consist of any data type (even more arrays), and are accessed by index numbers. Strings are read character-by-character, in contrast. Arrays can also be used to control flow, though I am not sure just how this works.
+Any list whose contents would need to be recalled individually could be stored in an array. 
 
 # What is one way, using Ruby, to retrieve the 6th element in an Array? How about the 8th element? What happens if you try to retrieve the value of the _9999th_ element (or some element that doesn't exist in the array)?
 
-Replace me with your response...
+Retrieving an element can be done with:
+
+arr = []
+arr[0] = "Jones"
+arr[1] = "Koresh"
+arr[2] = "Rajneesh"
+arr[3] = "Applewhite"
+puts arr [2]
+
+Because Ruby is zero-based indexed, to access the 6th element, use:
+
+puts arr [5]
+
+For the 8th element,
+
+puts arr [7]
+
+If we try to retrieve an element that doesn't exist in the array, we get an empty line.
 
 # The previous question asks about finding, for example, the 6th element in an Array. Is it possible to find the **-6th** (Notice the negative symbol!) element in an Array? What does that even mean?
 
-Replace me with your response...
+Yes, the -6th element in the array would be the 6th from the end, counting backwards with the last element being '1'.
 
 # How would you perform an operation on every element inside an Array?
 
-Replace me with your response...
+Using my above example, I put the cult leaders' names into lowercase:
+
+arr.each{ |x| puts x.downcase}      
 
 # How do you add elements to an Array?
 
-Replace me with your response...
+Simply, we can add elements to an array using:
+
+arr.push ("Jouret"),("Shinrikyo") 
```

Comment:
While this works, I'd like to caution you about the way you provided the new elements. Typically we would see your code like this instead:

```ruby
arr.push("Jouret", "Shinrikyo")
```

Ruby is somewhat forgiving about its syntax though, so your code also worked. No revision needed–just an FYI.